### PR TITLE
Improve a number of ghc-options checks

### PR DESCRIPTION
**Pull request by dcoutts**
In particular, allow flags that are appropriate during development if they're guarded by a manual flag.

Various other improvements, see the patch descriptions and diffs for details.

### CODE REVIEW

**Comment by 23Skidoo** [comment]
Fails on Travis due to redundant import:

```
Distribution/PackageDescription/Check.hs:39:1:
    Warning: The import of `Control.Applicative' is redundant
               except perhaps to import instances from `Control.Applicative'
             To import instances alone, use: import Control.Applicative()
```

**Comment by dcoutts** [comment]
How do we deal with the Control.Applicative versionitis issue in other modules?

**Comment by 23Skidoo** [comment]
@dcoutts `#ifdefs`. Travis helps there.

File: Cabal/Distribution/PackageDescription/Check.hs [hunk 1]
```diff
@@ -36,6 +36,7 @@ module Distribution.PackageDescription.Check (
 import Data.Maybe
          ( isNothing, isJust, catMaybes, maybeToList, fromMaybe )
 import Data.List  (sort, group, isPrefixOf, nub, find)
+import Control.Applicative
```
Comment (hvr) [inline Cabal/Distribution/PackageDescription/Check.hs]:
```
Distribution/PackageDescription/Check.hs:39:1: Warning:
    The import of ‘Control.Applicative’ is redundant
      except perhaps to import instances from ‘Control.Applicative’
    To import instances alone, use: import Control.Applicative()
```

assuming `Control.Applicative` is really needed, you can also just add an `import Prelude` after all `base`-imports

**Pull request merged**